sql/2022/third-parties/content_encoding.sql

```diff
@@ -0,0 +1,51 @@
+#standardSQL
+#content-encoding by third parties
+
+WITH requests AS (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    pageid AS page,
+    url,
+    resp_content_encoding AS content_encoding
+  FROM
+    `httparchive.summary_requests.2022_06_01_*`
+),
+
+third_party AS (
+  SELECT
+    domain,
+    COUNT(DISTINCT page) AS page_usage
+  FROM
+    `httparchive.almanac.third_parties` tp
+  JOIN
+    requests r
+  ON NET.HOST(r.url) = NET.HOST(tp.domain)
+  WHERE
+    date = '2022-06-01' AND
+    category != 'hosting'
+  GROUP BY
+    domain
+  HAVING
+    page_usage >= 50
+)
```

Comment:
Should we NET.HOST it here in the CTE to avoid any duplicates in the later join? Note sure if it makes a difference or if the thirdparties table is already basically at NET.HOST level but might be safer.

```suggestion
third_party AS (
  SELECT
    NET.HOST(domain) AS domain,
    COUNT(DISTINCT page) AS page_usage
  FROM
    `httparchive.almanac.third_parties` tp
  JOIN
    requests r
  ON NET.HOST(r.url) = NET.HOST(tp.domain)
  WHERE
    date = '2022-06-01' AND
    category != 'hosting'
  GROUP BY
    NET.HOST(domain)
  HAVING
    page_usage >= 50
)
```

Reply:
Updated the query and re-ran it. No changes.

Also I think the `NET.HOST(domain)` in the `GROUP BY` is incorrect since the `GROUP BY` uses the `SELECT` fields (so `domain` is already `NET.HOST`-ed).

Reply:
True!